src/main/java/dk/cngroup/lentils/service/HintService.java

```diff
@@ -0,0 +1,36 @@
+package dk.cngroup.lentils.service;
+
+import dk.cngroup.lentils.entity.Cypher;
+import dk.cngroup.lentils.entity.Hint;
+import dk.cngroup.lentils.repository.HintRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class HintService {
+
+    private HintRepository hintRepository;
+
+    @Autowired
+    public HintService(HintRepository hintRepository) {
+        this.hintRepository = hintRepository;
+    }
+
+    public List<Hint> getHintsForCypher(Cypher cypher) {
+        return hintRepository.findByCypher(cypher);
+    }
+
+    public Hint add(Hint hint) {
```

Comment:
Pouzivas v projektu `save` metodu, tak bych to drzet jednotne a menil bych to tady zase na `save`